v2/tools/generator/internal/config/glob_matcher.go

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package config
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+	"sync"
+
+	"github.com/Azure/azure-service-operator/v2/internal/set"
+	"github.com/pkg/errors"
+)
+
+// A globMatcher is used to match a string against a literal or wildcard
+type globMatcher struct {
+	glob       string         // The glob we're matching, may contain wildcards * and ?
+	regex      *regexp.Regexp // A regular expression to match the glob
+	once       sync.Once
+	matched    bool
+	candidates set.Set[string]
+}
+
+var _ StringMatcher = &globMatcher{}
+
+// newGlobMatcher returns a new matcher for handling wildcards
+func newGlobMatcher(glob string) *globMatcher {
+	// Guard against misuse
+	if !HasWildCards(glob) {
+		msg := fmt.Sprintf("glob string %q has no wildcards", glob)
+		panic(msg)
+	}
+
+	return &globMatcher{
+		glob:       glob,
+		candidates: make(set.Set[string]),
+	}
+}
+
+func (gm *globMatcher) Matches(term string) bool {
+	gm.once.Do(gm.createRegex)
+
+	if gm.regex.MatchString(term) {
+		if !gm.matched {
+			// First time we match, clear out our candidates as we won't be needing them
+			gm.matched = true
+			gm.candidates.Clear()
+		}
+
+		return true
+	}
+
+	if !gm.matched {
+		// Still collecting candidates
+		gm.candidates.Add(term)
+	}
+
+	return false
+}
+
+func (gm *globMatcher) WasMatched() error {
+	if gm.matched {
+		return nil
+	}
+
+	choices := set.AsSortedSlice(gm.candidates)
+	return errors.Errorf(
```

Comment:
Is this going to get ridiculously long in some cases?

Reply:
Potentially, yes - but I didn't see a good way to prune the list of candidates without potentially omitting the one the user wants.